<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<title>Untitled Document</title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
</head>

<body>
<p><strong>Supplemental Data for:</strong><br>
  Lewis et al., Cell 115, pp. 787&#8211;798<br>
</p>
<p><br>
  <a href="C0311075-lewis-figS1A.pdf">Figure S1A.</a> The locations and 
  secondary structures of the TargetScan predicted miRNA binding sites are diagrammed 
  for the 115 predicted vertebrate targets (representing 107 distinct genes) conserved 
  between human, mouse, rat, and Fugu. For each conserved miRNA-UTR pair, the 
  name of the miRNA, the names of the orthologous target mRNAs in each species, 
  and the Ensembl description for the human gene are shown. The next line contains 
  the length of the sequence used as the human UTR and the positions of the 5&#8217; 
  ends of the seed matches, followed by schematic of the UTR where each dash represents 
  25 nt and the location of each seed match is represented by a &#8220;1.&#8221; 
  Below the UTR schematic are the TargetScan generated duplex structures used 
  to score the UTR. Each sequence consists of the miRNA sequence joined to the 
  binding site by a &#8220;&amp;,&#8221; with the predicted base pairing indicated 
  by the parentheses below the sequence. Calculated free energies of binding for 
  the duplexes (in kcal/mol) are shown next to the structures. Schematics, structures, 
  and energies are shown for all orthologous genes considered by TargetScan in 
  predicting these targets.<br>
  <br>
  <a href="C0311075-lewis-figS1B.pdf">Figure S1B.</a> The locations and 
  secondary structures of the TargetScan predicted miRNA binding sites are diagrammed 
  for the 854 predicted mammalian targets (representing 442 distinct genes) of 
  rMamm miRNAs conserved between human, mouse, and rat. The format is similar 
  to that of Figure S1A.<br>
  <br>
  <a href="C0311075-lewis-figS2.pdf">Figure S2.</a> The locations and secondary 
  structures of predicted miRNA binding sites are diagramed for the 13 tested 
  human genes and their mammalian orthologs, as well as the lin-41 control, using 
  the format of Figure S1A. The portion of the human UTR that was PCR amplified 
  and cloned downstream of the luciferase reporter is highlighted in blue in the 
  UTR schematic. The sequences of the wild type and mutant PCR products are indicated 
  at the end of each record. Seed matches are shown in blue for the wild-type 
  sequence and the corresponding sequences are highlighted in red for the mutant 
  sequence, with mutated residues shown in lowercase. The last record shows sequences 
  of the lin-41 controls (lin-41, a known target of C. elegans let-7 RNA) referred 
  to in Figure 3, with regions of complementarity to let-7 RNA underlined and 
  mutations shown in red lowercase letters.<br>
  <br>
  <a href="C0311075-lewis-TableS1.xls">Table S1.</a> The Human miRNAs and 
  Shuffled Controls used in This Study<br>
  The inclusion of each miRNA in the three subsets used in this study (rMamm, 
  nrMamm, and nrVert) is indicated by Y (Yes) or N (No). For those miRNAs in nrMamm, 
  the sequences of the four shuffled variants generated my MiRhuffle are listed 
  on the next four lines (labeled miR-X_sh0, miR-X_sh1, etc.). The 19 miRNAs not 
  in rMamm are those for which Fugu homologs could not be identified. For completeness, 
  we list here a few additional human miRNAs that were not used in this study 
  for various reasons, such as imperfect conservation between human and mouse: 
  miR-17, miR-103b, miR-129, miR-134, miR-150, miR-151, miR-189 miR-200a, miR-217. 
  <br>
  <br>
  <a href="C0311075-lewis-TableS2.xls">Table S2.</a> The Predicted Mammalian 
  Targets for miRNAs in rMamm<br>
  The 442 genes in this set were predicted as targets of rMamm miRNAs by TargetScan 
  in human, mouse, and rat orthologs, as described in the text. MiRNAs with identical 
  seeds that are predicted to target the same gene are shown in a single row of 
  the table. MiRNAs with different seeds that target the same gene are listed 
  on separate lines.<br>
  <br>
  <a href="C0311075-lewis-TableS3.xls">Table S3.</a> Targets of Shuffled 
  Control Sequences<br>
  The 558 shuffled sequence:UTR pairs found human, mouse, and rat that were predicted 
  by TargetScan for any of the four cohorts of MiRshuffled variants of nrMamm 
  miRNAs are shown.<br>
  <br>
  <a href="C0311075-lewis-TableS4.xls">Table S4.</a> Biological Function 
  Classes of Predicted miRNA Targets and Controls<br>
  The number and percentage of UTRs annotated in various Gene Ontology biological 
  process categories are shown for the 400 predicted miRNA-UTR pairs predicted 
  by TargetScan for nrMamm miRNAs; the miRNA-UTR pairs predicted with randomized 
  miRNAs (average of 4 cohorts); and for the total set of orthologous genes conserved 
  between human, mouse, and rat. For cases in which GO categories with a parent-child 
  relationship are shown, the child is indented. Note that the GO categories are 
  not mutually exclusive.<br>
  <br>
  <a href="C0311075-lewis-TableS5.xls">Table S5.</a> The Predicted Vertebrate 
  Targets for miRNAs in nrVert<br>
  The orthologous genes for this set scored highly as targets of nrVert miRNAs 
  in all four organisms studied. MicroRNAs with different seeds that target the 
  same gene are listed on separate lines. </p>
</body>
</html>
